Sources/PickBetter/SegmentedBetterPickerStyle.swift

```diff
@@ -0,0 +1,170 @@
+    ) {
+        self.frameWidth = frameWidth
+        self.frameHeight = frameHeight
+        self.horizontalCellAlignment = horizontalCellAlignment
+    }
+
+    /// Builds the overall picker ``View``.
+    public func makeView(_ configuration: Configuration) -> some View {
+        /// Full width of the `View` for a cell.
+        let cellWidth = (frameWidth / CGFloat(configuration.cellCount)) - 1.5
+        /// Index of the selected cell.
+        let selectCellIndex = configuration.selectionIndexSet.first
+
+        ZStack(alignment: .leading) {
+            if let selectCellIndex = selectCellIndex {
+                /// Rounded background to display selected item
+                RoundedRectangle(cornerRadius: 6)
+                    .fill(.white)
+                    .frame(width: cellWidth, height: frameHeight)
+                    .offset(x: CGFloat(selectCellIndex) * cellWidth)
+                    .animation(.easeInOut(duration: 0.25), value: selectCellIndex)
+                    .padding(2)
+                    .shadow(color: .secondary.opacity(0.3), radius: 2, y: 1)
+            }
+
+            HStack {
+                configuration.listOutput
+            }
+            .foregroundStyle(.black)
+            .frame(width: frameWidth, height: frameHeight)
+        }
+        .background(Color(red: 0.92, green: 0.92, blue: 0.92))
+        .clipShape(RoundedRectangle(cornerRadius: 8))
+    }
+
+    /// Builds each individual cell ``View``.
+    public func makeListCell(_ configuration: CellConfiguration) -> some View {
+        Button(action: {
+            configuration.select()
+        }) {
+            HStack {
+                if [.center, .trailing].contains(horizontalCellAlignment) {
+                    Spacer()
+                }
+
+                configuration.label()
+                    .padding(.horizontal, 6)
+
+                if [.center, .leading].contains(horizontalCellAlignment) {
+                    Spacer()
+                }
+            }
+        }
+        .contentShape(Rectangle()) // Used to make the entire HStack tappable
+    }
+}
+
+#if DEBUG
+    private struct PreviewItem: Identifiable, Sendable {
+        let id: String
+        let imageString: String
+    }
+
+    private let items: [PreviewItem] = [
+        .init(id: "Label 1", imageString: "list.bullet"),
+        .init(id: "Label 2", imageString: "list.bullet"),
+        .init(id: "Label 3", imageString: "list.bullet"),
+    ]
+
+    private func itemContent(_ item: PreviewItem) -> some View {
+        Image(systemName: item.imageString)
+            .resizable()
+            .frame(width: 16, height: 12)
+    }
+
+    private func itemContentWithLabel(_ item: PreviewItem) -> some View {
+        HStack {
+            Text(item.id)
+            Image(systemName: item.imageString)
+                .resizable()
+                .frame(width: 16, height: 12)
+        }
+    }
+
+    struct SegmentedBetterPickerStyle_Previews: PreviewProvider {
```

Comment:
Would it be better to use `#Preview` macro? This works but I don't know if the new API gives any benefits.

Reply:
I could switch it over, but I'm not sure there is a huge benefit. `#Preview` macro was introduced in Swift 5.9 and I think there are some issue running it via iOS 17 or lower (This may not be an issue because it's only in the simulator). The only real benefit I see is the ability to add a title to the Preview and the condensed code. I'm open to changing it, just wanted to give me reasoning.